src/views/products/ProductDetails.js

```diff
@@ -55,6 +43,38 @@ const getStockBadge = stock => {
   );
 };
 
+const respondToCartStatus = (status, isLoggedIn, productName, addMessage, setPurchasedTrue) => {
```

Comment:
So we're making a request to the backend (line 23) before checking the isLoggedIn boolean? 
If the user isn't logged in, we'd be making an unnecessary request. I think it would be better to do the isLoggedIn check before